src/typescript-service.ts

```diff
@@ -356,6 +363,39 @@ export class TypeScriptService implements LanguageHandler {
 		return deps;
 	}
 
+	async getCompletions(params: TextDocumentPositionParams): Promise<CompletionList> {
+		const uri = util.uri2reluri(params.textDocument.uri, this.root)
+		const line = params.position.line;
+		const column = params.position.character;
+		const fileName: string = util.uri2path(uri);
+
+		await this.projectManager.ensureFilesForReferences(uri);
+
+		const configuration = this.projectManager.getConfiguration(fileName);
+		await configuration.ensureAllFiles();
+
+		const sourceFile = this.getSourceFile(configuration, fileName);
+		if (!sourceFile) {
+			return null;
+		}
+
+		const offset: number = ts.getPositionOfLineAndCharacter(sourceFile, line, column);
+		const completions = configuration.getService().getCompletionsAtPosition(fileName, offset);
+
+		if (completions == null) {
+			return null;
+		}
+
+		return CompletionList.create(completions.entries.map((item) => {
+			const ret = CompletionItem.create(item.name);
+			const kind = completionKinds.get(item.kind);
+			if (kind) {
+				ret.kind = kind;
+			}
```

Comment:
Pretty sure you can just set the return value directly. Still a lot more properties though, documentation, type etc